cohortextractor2/query_engines/spark.py

```diff
@@ -57,6 +57,9 @@ def get_temp_table_name(self, table_name):
         self._temp_table_names.add(temp_table_name)
         return temp_table_name
 
+    def get_temp_database(self):
+        return self.backend.temporary_database
```

Comment:
Does this attribute exist on the backend? I can't see it being created anywhere.

I'm also not sure whether it's a property of the backend or a property of the connection.I can see arguments either way. But as things are currently conceived the backend defines the schema but it doesn't know where that schema lives; that's defined by the environment at runtime. I feel like the temporary database name belongs in the same bucket.

Reply:
It's set in the base backend: https://github.com/opensafely-core/cohort-extractor-v2/blob/main/cohortextractor2/backends/base.py#L23

Reply:
I just used the existing config infrastructure for the TPP db's use of a temporary database, basically

Reply:
Ahhh, fine yes, sorry! It's because I'd forgotten how the config gets passed in to the backend constructor. That could all do with refactoring at some point, but not now.